include/swift/Frontend/Frontend.h

```diff
@@ -433,6 +447,57 @@ class CompilerInstance {
   /// Frees up the ASTContext and SILModule objects that this instance is
   /// holding on.
   void freeContextAndSIL();
+
+private:
+  /// Load stdlib & return true if should continue, i.e. no error
+  bool loadStdlib();
+  ModuleDecl *importUnderlyingModule();
+  ModuleDecl *importBridgingHeader();
+
+  void
+  getImplicitlyImportedModules(SmallVectorImpl<ModuleDecl *> &importModules);
+
+public: // for static functions in Frontend.cpp
+  struct ImplicitImports {
+    SourceFile::ImplicitModuleImportKind kind;
+    ModuleDecl *objCModuleUnderlyingMixedFramework;
+    ModuleDecl *headerModule;
+    SmallVector<ModuleDecl *, 4> modules;
+
+    explicit ImplicitImports(CompilerInstance &compiler);
+  };
+
+private:
+  void createREPLFile(ImplicitImports &implicitImports) const;
+  std::unique_ptr<DelayedParsingCallbacks> computeDelayedParsingCallback();
+
+  void addMainFileToModule(ImplicitImports &implicitImports);
+
+  void parseAndCheckTypes(ImplicitImports &implicitImports);
+
+  void parseLibraryFile(unsigned BufferID, ImplicitImports &implicitImports,
+                        PersistentParserState &PersistentState,
+                        DelayedParsingCallbacks *DelayedParseCB);
+
+  /// Return true if had load error
+  bool
+  parsePartialModulesAndLibraryFiles(ImplicitImports &implicitImports,
+                                     PersistentParserState &PersistentState,
+                                     DelayedParsingCallbacks *DelayedParseCB);
+
+  void checkTypesWhileParsingMain(PersistentParserState &PersistentState,
+                                  DelayedParsingCallbacks *DelayedParseCB);
```

Comment:
I'm not sure if I can come up with anything better, but having both `checkTypesWhileParsingMain` and `parseAndTypeCheckMainFile` seems suboptimal. I suppose you could get rid of this level and just call lift the contents of this function back into `parseAndCheckTypes`? It doesn't look like separating them is providing any additional clarity just because of how much is going on in `checkTypesWhileParsingMain`.

Reply:
Fixed as per our offline discussion.